test/functional/connections_survive_primary_step_down_spec_tests.js

```diff
@@ -0,0 +1,299 @@
+'use strict';
+
+const chai = require('chai');
+const expect = chai.expect;
+
+let client;
+describe('Connections survive primary step down', function() {
+  beforeEach(function() {
+    client = this.configuration.newClient(
+      { w: 1 },
+      { poolSize: 1, retryWrites: false, useUnifiedTopology: true }
+    );
+  });
+
+  afterEach(function() {
+    return client.close();
+  });
+
+  it('getMore iteration', {
+    metadata: { requires: { mongodb: '>=4.2.0', topology: 'replicaset' } },
+    test: function(done) {
+      client.connect(err => {
+        expect(err).to.not.exist;
+        const db = client.db('step-down');
+
+        db.admin().serverStatus((err, result) => {
```

Comment:
Could we refactor these tests to use the Promise API instead of using `done`?

```js
const db = client.db('step-down');
let numberOfConnections;
let collection;

return Promise.resolve()
  .then(() => db.admin.serverStatus())
  .then(result => {
    numberOfConnections = result.connections.totalCreated;
  })
  .then(...)
  ...
```

It would remove a lot of the indentation and the `expect(err).to.not.exist`s.